src/components/common/text/CryptoText.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+
+import { useCryptoText } from '../../../hooks/useCryptoText'
+import { getDisplayDenominationFromState, getExchangeDenominationFromState } from '../../../selectors/DenominationSelectors'
+import { useDispatch, useSelector } from '../../../types/reactRedux'
+import { fixFiatCurrencyCode, getDenomFromIsoCode, zeroString } from '../../../util/utils'
+
+type Props = {
+  walletId: string,
+  tokenId?: string,
+  nativeAmount?: string
+}
+
+/**
+ * Returns a cleaned crypto amount string. If no nativeAmount is given, use the
+ * wallet balance.
+ **/
+export const CryptoText = ({ walletId, tokenId, nativeAmount }: Props) => {
+  const dispatch = useDispatch()
+  const currencyInfo = useSelector(state => state.core.account.currencyWallets[walletId].currencyInfo)
```

Comment:
You cannot use `useSelector` to grab things out of the account like this. You have to do it this way:

```
const account = useSelector(state => state.core.account)
const wallet = account.currencyWallets[walletId]
const { currencyInfo, fiatCurrencyCode } = wallet
```

This is because the properties of the account are *not* managed by redux, so it won't know when they change or when to re-render. In this particular case, both these items are fixed. In other cases, such as balances, the wallet can update them at any time, so you actually need to use a subscription and `useState`.